### ISSUE TITLE
Cassandra node process is terminated when there is an issue with Cassandra-exporter agent

### ISSUE THREAD

**Issue by podile** [instaclustr/cassandra-exporter#68]
Cassandra node process is terminated when cassandra-exporter agent fails to initialize. The impact may be less for smaller Cassandra environments, but the impact is more on bigger Cassandra environments. Agent should not terminate the node process if there any error during initialization. Introduce a new flag based initialization logic to drop agent initialization in case of any error allow the node process to continue.

**Comment by podile** [comment]
I am going to work in this enhancement.

**Comment by podile** [comment]
I will work on this enhancement.

**Comment by zegelin** [comment]
I disagree. The agent is an extension to Cassandra. Any error in the Cassandra config prevents Cassandra from starting. Likewise, any config error with the exporter agent should follow the same pattern.

**Comment by podile** [comment]
I completely agree that patterns wise the process should get terminated in case of any issue with config, and that should be the default behavior.  Adding a flag (--dont_exit_on_error) to continue the node process if agent could't initialize will give peace of mind to the users who maintain huge clusters.

**Comment by zegelin** [comment]
What kind of errors are we talking about here? Configuration problems should abort IMO.

**Comment by podile** [comment]
Configuration problems should abort cassandra-exporter agent initialization, should the node process be aborted if Casandra-exporter fails to initialize?

**Comment by zegelin** [comment]
Hi @podile. I now have a better understanding of the need for this feature and happy for abort-on-config-problems to be put behind a flag.
